RichString.c

```diff
@@ -20,18 +20,25 @@ in the source distribution for its full text.
 #define charBytes(n) (sizeof(CharType) * (n))
 
 static void RichString_extendLen(RichString* this, int len) {
-   if (this->chlen <= RICHSTRING_MAXLEN) {
+   if (this->chptr == this->chstr) {
+      // String is in internal buffer
```

Comment:
Maybe add the old condition as an assert here (`assert(this->chlen <= RICHSTRING_MAXLEN);`)?

Reply:
Done.